DataFormats/Reconstruction/src/TrackFwd.cxx

```diff
@@ -378,11 +344,7 @@ void TrackExtrap::addMCSEffect(TrackParamMFT* trackParam, double dZ, double x0)
   sigmathetasq *= sigmathetasq * pathLengthOverX0;
```

Comment:
Is x0 the radiation length or its fraction. If you assume that the scattering is point-like, then the dZ itself should be irrelevant (except in the definition of the fractional x/x0). In any case, the line  https://github.com/AliceO2Group/AliceO2/blob/0fb0bb5df234ff5609316e8022e224a81fd8a7a5/DataFormats/Reconstruction/src/TrackFwd.cxx#L343 misses the ``sqrt(x/x0)`` term. Also, here one should use not 1/pt but 1/pz, as this is the component normal to the scattering plane.
Note, that the log. term makes the calculated error^2 non-linear over the x/x0, which is a bad feature if you do multiple steps in a thick scattered. In the TrackParCov we suppress this log term.

Also, what make the MS treatment such asymmetric in dZ sign?

Reply:
Hi @shahor02 , 

Thanks for your comments and spotting this. It should be 1/p (total momentum), not pt nor pz. As for the log term, it's suppression would be a valid approximation only when the argument is close to 1. I see that for small x/X0 = 0.1% the suppression of the log leads to up to 35% difference in the standard deviation of the MCS dispersion. Isn't this worst than the non-linear error?

The sqrt(x/x0) term is added (squared) on the line bellow. 

The asymmetry on dZ sign is a trick inherited from MCH tracking to choose between MCS evaluated point-like (dZ <= 0) or point-like followed by a linear propagation by a distance dZ (dZ >0). We are using point-like only, thus I will remove the later.

Reply:
Hi @rpezzi 

The MS adds a pt-kick in the direction normal to particle momentum, hence the angular deflection pt-kick/p in the track comoving frame. Are your modifications of the cov. matrix properly project this deflection in the comoving frame to global?
For the log term: is your x/x0 ~0.1% or order of magnitude larger? The recipe should depend on how you apply the correction. If you are doing small steps each time correcting for very thin slice, then it is better to omit this term and have instead error^2 additive, if you have just 1 very thin slice, then this term may help. In the barrel we don't use this term.

Reply:
Hi @shahor02 , 
For the MFT, MCS increment on the Cov matrix is added only once per tracklet, when it crosses one or more MFT disks, omitted otherwise. Each disk accounts for about x/X0 = 0.8% - where the log term contributes with 22% of scattering dispersion. On the other hand, accumulating the material effects at the beginning of the track overestimates its effects, this has not been quantified yet.. 

The translation between track->global reference frame is accounted for in these calculations.

Reply:
ok, you should probably try with and w/o log term and see which one gives better pulls.

Reply:
Hi @shahor02 , your comments have been addressed with the commits pushed today. MC labels are now computed and stored like other detectors. Same for cluster indices. Tracking classes and redundant data have been removed from reco output file.  Fine tuning MCS parameters is yet to be done in a forthcoming PR.